src/main/java/com/madgag/gif/fmsware/AnimatedGifEncoder.java

```diff
@@ -345,6 +372,30 @@ protected int findClosest(Color c) {
 	}
 	
 	/**
+	 * Returns index of palette color closest to c
+	 *
+	 */
+	protected int findExact (Color c) {
+		if (colorTab == null) return -1;
+		int r = c.getRed() & 0xff;
+		int g = c.getGreen() & 0xff;
+		int b = c.getBlue() & 0xff;
+		int len = colorTab.length;
+		for (int i = 0; i < len; i++) {
+			int index = i / 3;
+			if (usedEntry[index]) {
+				int tr = colorTab[i++] & 0xff;
+				int tg = colorTab[i++] & 0xff;
+				int tb = colorTab[i] & 0xff;
+				if (r==tr && g==tg && b==tb) {

```

Comment:
I have also added a new public method isColorUsed, which returns true if the exact matching color is existing, and used in the color palette, otherwise, return false. This method has to be called before
finishing the image, because after finished the palette is destroyed and it will always return false.